torch/testing/_internal/distributed/distributed_test.py

```diff
@@ -5119,3 +5119,72 @@ def test_output_unused_in_loss(self):
                                 )
                         else:
                             self.assertFalse(True, "DDP error not raised")
+
+        @require_backend({"gloo"})
+        @require_backends_available({"gloo"})
+        def test_monitored_barrier_gloo(self):
+            process_group = dist.new_group(ranks=list(i for i in range(int(self.world_size))))
+            tensors = [torch.ones(10) * self.rank]
+            # Kick off some allreduce work on all ranks
+            for _ in range(10):
+                process_group.allreduce(tensors).wait()
+            # Run monitored barrier
+            timeout = timedelta(seconds=2)
```

Comment:
nit: maybe we can use `timeout=0.1` everywhere to ensure the test runs faster?

Reply:
Sounds good, though if we use too small of a timeout, monitored_barrier() in cases where it's expected to succeed may fail, since there is some latency in the interprocess comm.